common/src/main/java/io/thorntail/openshift/ts/common/http/HttpUtil.java

```diff
@@ -0,0 +1,36 @@
+package io.thorntail.openshift.ts.common.http;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+
+public class HttpUtil {
+
+    public static Map<String, HttpClientResponse> httpRequests(String url, int numberOfThreads, int sleepMillis)
+            throws Exception {
+        Map<String, HttpClientResponse> responses = new HashMap<>();
+        PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager();
+
+        HttpClientThread[] threads = new HttpClientThread[numberOfThreads];
+        for (int i = 0; i < threads.length; i++) {
+            CloseableHttpClient httpClient = HttpClients.custom().setConnectionManager(cm).build();
+            HttpGet httpget = new HttpGet(url);
+            threads[i] = new HttpClientThread(httpClient, httpget);
+        }
+
+        for (int i = 0; i < threads.length; i++) {
+            threads[i].start();
+            Thread.sleep(sleepMillis);
```

Comment:
The entire approach looks kinda ugly, but I thought that perhaps you need to run requests truly in parallel, so that the load balancer in OpenShift really does round robin. And then I noticed this -- after each thread/request is started, you wait 500 ms. That is long enough to make sure that when new thread is started, the old one already finished. So there's no real parallelism, and all the dance around threads is just fake.